DotNet UWP Samples/Earthquake Simulator/Earthquake Simulator/App.xaml.cs

```diff
@@ -0,0 +1,309 @@
+        {
+            throw new Exception("Failed to load Page " + e.SourcePageType.FullName);
+        }
+
+        /// <summary>
+        /// Invoked when application execution is being suspended.  Application state is saved
+        /// without knowing whether the application will be terminated or resumed with the contents
+        /// of memory still intact.
+        /// </summary>
+        /// <param name="sender">The source of the suspend request.</param>
+        /// <param name="e">Details about the suspend request.</param>
+        private void OnSuspending(object sender, SuspendingEventArgs e)
+        {
+            var deferral = e.SuspendingOperation.GetDeferral();
+
+            deferral.Complete();
+        }
+
+        protected override void OnBackgroundActivated(BackgroundActivatedEventArgs args)
+        {
+            base.OnBackgroundActivated(args);
+            IBackgroundTaskInstance taskInstance = args.TaskInstance;
+            AppServiceTriggerDetails appService = taskInstance.TriggerDetails as AppServiceTriggerDetails;
+            _appServiceDeferral = taskInstance.GetDeferral();
+            taskInstance.Canceled += OnAppServicesCanceled;
+            DataStreamerConnection = appService.AppServiceConnection;
+            DataStreamerConnection.RequestReceived += OnAppServiceRequestReceived;
+            DataStreamerConnection.ServiceClosed += AppServiceConnection_ServiceClosed;
+        }
+
+        private void OnAppServicesCanceled(IBackgroundTaskInstance sender, BackgroundTaskCancellationReason reason)
+        {
+            _appServiceDeferral.Complete();
+        }
+
+        /// <summary>
+        /// Connection with DataStreamer has been severed
+        /// </summary>
+        private async void AppServiceConnection_ServiceClosed(AppServiceConnection sender, AppServiceClosedEventArgs args)
+        {
+            await _service.Disconnect();
+
+            try
+            { 
+                await this.ViewModel.SetDataStreamerStatus("Not Connected");
+                await this.ViewModel.SetDataStreamerAppVersion("");
+                await this.ViewModel.SetDataStreamerApiVersion("");
+            }
+            catch(Exception ex)
+            {
+            }
+
+            _appServiceDeferral.Complete();
+       }
+
+        ///
+        private async void OnAppServiceRequestReceived(AppServiceConnection sender, AppServiceRequestReceivedEventArgs args)
+        {
+            // Get a deferral because we use an awaitable API below to respond to the message
+            // and we don't want this call to get cancelled while we are waiting.
+            var messageDeferral = args.GetDeferral();
+            ValueSet message = args.Request.Message;
+            ValueSet returnData = new ValueSet();
+
+            string command = message["Command"] as string;
+
+            try
+            {
+                switch (command)
+                {
+                    case "Connect":
```

Comment:
What does each case imply? Code comments

Reply:
I think  each case is obvious